Src/Test.UnitTests.Sarif.PatternMatcher/PatternInvariantTests.cs

```diff
@@ -51,22 +60,85 @@ public static void VerifyAllValidatorsExist(string definitionsFileName)
                         string.Join($",{Environment.NewLine}  ", rulesWithoutValidators));
         }
 
-        public static void VerifyAllSharedStringsExist(string definitionsFilePath, string sharedStringsFilePath)
+        public static void VerifyAllRuleFilenamesMatchDefinitions(string definitionsFilePath)
         {
+            //Read the json file content to get the rule names
             string content = File.ReadAllText(definitionsFilePath);
+            SearchDefinitions sdObject = JsonConvert.DeserializeObject<SearchDefinitions>(content);
+            var regexSet = new HashSet<string>();
+            foreach (SearchDefinition searchDefinition in sdObject.Definitions)
+            {
+                foreach (MatchExpression matchExpression in searchDefinition.MatchExpressions)
+                {
+                    regexSet.Add(matchExpression.Name.Split('/')[1]);
+                }
+            }
 
-            var jObject = (JObject)JToken.Parse(content);
+            // Load up all Validator files in plugin directory
+            var definitionsDirectory = new DirectoryInfo(definitionsFilePath);
+            string definitionsParentDirectory = definitionsDirectory.Parent.FullName;
+            string validatorsDirectory = Path.Combine(definitionsParentDirectory, "SecurePlaintextSecretsValidators");
+            var validatorsDirectoryInfo = new DirectoryInfo(validatorsDirectory);
 
-            IEnumerable<string> rules = jObject["Definitions"][0]["MatchExpressions"].Select(x => x["ContentsRegex"].ToString()).Distinct();
+            // load up all Test files in Test.plugin directory
```

Comment:
>load [](http://example.com/codeflow?start=15&length=4)

Please capitalize all comments. Make them complete sentences. End them with periods. I'm merging all this, you can clean things up as you continue to work. :)